<launch>
    <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen">
        <!-- Remap topic required by the node to custom topics -->
        <remap from="image_rect" to="/imageResize/output_video" />
        <remap from="camera_info" to= "/imageResize/camera_info" />

        <!-- Optional: Subscribe to the compressed stream-->
        <param name="image_transport" type="str" value="raw" />

        <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
        <param name="tag_family" type="str" value="36h11" />

        <!-- Enable projected optical measurements for more accurate tag transformations -->
        <!-- This exists for backwards compatability and should be left true for new setups -->
        <param name="projected_optics" type="bool" value="true" />

        <!-- Describe the tags -->
        <rosparam param="tag_descriptions">[
            {id: 10, size: 0.2, frame_id: cage_tag}]
        </rosparam>
    </node>
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
        <param name="video_device" value="/dev/video0" />
        <param name="image_width" value="320" />
        <param name="image_height" value="240" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="usb_cam" />
        <param name="io_method" value="mmap"/>
        <param name="autoexposure" value="true"/>
        <param name="framerate" value="30"/>
        <param name="sharpness" value="255"/>
    </node>
    <node name="resizeCameraInfo" pkg="april_drone3" type="resizeCameraInfo" >
    </node>
    <node pkg="image_view" type="image_view" name="original_viewer">
        <remap from="image" to="/usb_cam/image_raw"/>
        <param name="autosize" type="bool" value="true"/>
    </node>
    <node pkg="image_view" type="image_view" name="resize_img_viewer">
        <remap from="image" to="imageResize/output_video"/>
        <param name="autosize" type="bool" value="true"/>
    </node>

</launch>
